lessons/advanced/typespec.md

```diff
@@ -0,0 +1,197 @@
+---
+layout: page
+title: Specifications and types
+category: advanced
```

Comment:
I think if we will move the `Dialyxir/Dialyzer` part to another lesson in the `specifics` section the remaining could still be a part of `basics`.